src/lib/pricingEngine.js

```diff
@@ -0,0 +1,64 @@
+const { BigNumber } = require('ethers');
+
+const { calculateProductDataForTranche } = require('./helpers');
+const { selectProductPools, selectProduct } = require('../store/selectors');
+
+/**
+ * Calculates the pricing information for a given product based on its associated pools,
+ * weighted by each pool's available capacity.
+ *
+ * @param {Object} store - The application state store containing product data.
+ * @param {number} productId - The unique identifier for the product to calculate pricing for.
+ * @returns {Object|null} An object containing the product ID, an array of price per pool,
+ *                       and the weighted average price, or null if no pools are found.
+ */
+function pricingEngine(store, productId) {
+  const { assets, assetRates } = store.getState();
+  const product = selectProduct(store, productId);
+  const productPools = selectProductPools(store, productId);
+  const now = BigNumber.from(Math.floor(Date.now() / 1000));
+
+  if (!product || !productPools.length) {
+    return null;
+  }
+
+  const pricePerPool = [];
+  let totalWeight = BigNumber.from(0);
+  let weightedSum = BigNumber.from(0);
+
+  // Get capacity data per pool
+  const { capacityPerPool } = calculateProductDataForTranche(
+    productPools,
+    0, // use trancheIndex 0 to include all active tranches in pool capacity calculation
```

Comment:
this is the firstUsableTrancheIndex based on the product period + gracePeriod. I'm hardcoding 0 here so that we include all active tranche in capacity calculation

NOTe: The available capacity of the pool is only used as weight for calculating the weighted average of the product price across all pools.